### DISCUSSION TITLE
What to do with material_theme_project_new.xml?

### DISCUSSION THREAD

**Discussion by charliefoxtwo** [AtomMaterialUI/material-theme-issues#506] [Q&A] [answered]
Since the latest update (9.0.0 I think?), a `material_theme_project_new.xml` file seems to be created with every project of mine. I'm trying to understand the purpose of this file - should it be committed to the repo? If so, how does it work if I'm working in a repo with multiple other individuals, and what if they also have the plugin? If not, do I now have to add this to the gitignore of every repo I work in? Either way, what should I do with repos which I contribute to but don't own - it feels like it would be overstepping to either contribute a random config file or add a random config file to the .gitignore.

**Comment by mallowigi** [comment]
This file is your material theme settings. There are two files: global settings in your IDE configuration folder, and project settings for each project. You don't have to commit it unless you want to share your per-project settings to your coworkers or something (but usually this is more like a user preference so I don't recommend sharing it)

**Reply by charliefoxtwo** [reply, marked as answer]
Got it - so basically I need to add this to the .gitignore of every project I work in now? The default /.idea/.gitignore doesn't seem to cover it and neither do any of the standard jetbrains .gitignores

**Reply by charliefoxtwo** [reply]
I know this seems like a weird hill to die on, but is it possible to just roll back to v8 which didn't have this settings file? I don't like the idea of adding a config file for a personal plugin to the gitignore of every repo I work in. Every time I delete the file or stash it, it seems to just be regenerated, so it doesn't seem like it's even needed (at least not for my use case)?

**Reply by mallowigi** [reply]
No, this is out of the question. There are many other files added to the .idea folder coming from other plugins as well, no reason this one should be an exception. What do you do for these files?

**Reply by charliefoxtwo** [reply]
I have never had any other plugin file appear in my project. The plugins I use must either not generate them, or must put them somewhere that is already gitignored by the standard jetbrains gitignore. Could you provide an example of such a file or plugin?

To be clear, I'm asking if _I_ can roll back in some way - not if the entire plugin can roll back 😄

**Reply by mallowigi** [reply]
Well if you look in the .idea folder you should see plenty of other settings files. Example from my project:

![CleanShot 2024-05-13 at 19 03 20@2x](https://github.com/mallowigi/material-theme-issues/assets/5015756/fa6868c5-0032-42e3-ab65-cab428dc6490)

I myself put the whole .idea folder in gitignore usually, so I never have this problem

**Reply by charliefoxtwo** [reply]
Jetbrains seems to advise that some of the files in the .idea folder should be committed (https://rider-support.jetbrains.com/hc/en-us/articles/207097529-What-is-the-idea-folder) and recommends using the following gitignore, which is what I use in my projects: https://github.com/github/gitignore/blob/main/Global/JetBrains.gitignore

That gitignore does appear to have specific carveouts for a few larger plugins. I wonder if it would be possible to submit a PR to add material theme's new config file as well?

**Reply by mallowigi** [reply]
I'm not fond of the idea. But feel free to do so yourself, you don't need my permission :)
